rcl/src/rcl/publisher.c

```diff
@@ -205,6 +205,15 @@ rcl_publisher_init(
   goto cleanup;
 fail:
   if (publisher->impl) {
+    if (publisher->impl->rmw_handle) {
+      if (rmw_destroy_publisher(
+          rcl_node_get_rmw_handle(node),
+          publisher->impl->rmw_handle) != RMW_RET_OK)
+      {
```

Comment:
I'd recommend this style instead:

```suggestion
      rmw_ret_t rmw_fail_ret = rmw_destroy_publisher(
        rcl_node_get_rmw_handle(node), publisher->impl->rmw_handle);
      if (RMW_RET_OK != rmw_fail_ret)
      {
```